.gitlab/notify/notify.yml

```diff
@@ -163,3 +163,28 @@ notify_failure_summary_daily:
         echo 'Sending weekly summary'
         inv -e notify.failure-summary-send-notifications --weekly-summary
       fi
+
+close_failing_tests_stale_issues:
+  stage: notify
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/deb_x64$DATADOG_AGENT_BUILDIMAGES_SUFFIX:$DATADOG_AGENT_BUILDIMAGES
+  rules:
+    # Daily
+    - if: $CI_COMMIT_BRANCH != "main" || $CI_PIPELINE_SOURCE != "schedule"
+      when: never
+    - !reference [.on_deploy_nightly_repo_branch_always]
+  needs: []
+  tags: ["arch:amd64"]
+  script:
+    - weekday="$(date --utc '+%A')"
+    # Weekly on Friday
+    - |
+      if [ "$weekday" != "Friday" ]; then
+        echo "This script is run weekly on Fridays"
+        exit
+      fi
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
Have you looked if this can be a `rule` rather than a script? Not sure if there is a `DAY_OF_THE_WEEK` env var in Gitlab

Reply:
Unfortunately there is no such variable, I haven't found a way to do it within the `rule` section